app/code/Magento/CatalogInventory/Model/StockStateProvider.php

```diff
@@ -119,13 +119,13 @@ public function checkQuoteItemQty(StockItemInterface $stockItem, $qty, $summaryQ
         $result->setItemIsQtyDecimal($stockItem->getIsQtyDecimal());
         if (!$stockItem->getIsQtyDecimal()) {
             $result->setHasQtyOptionUpdate(true);
-            $qty = (int) $qty;
+            $qty = (int) $qty ?: 1;
             /**
              * Adding stock data to quote item
              */
             $result->setItemQty($qty);
             $qty = $this->getNumber($qty);
```

Comment:
I also have a small concern regarding this line.
```
$qty = $this->getNumber($qty);
```
It looks like a typo here, doesn't it?
Maybe it was expected to use `$origQty` instead of `$qty`, but we need to check this point.